addons/website_mail_channel/static/src/snippets/s_channel/options.js

```diff
@@ -8,12 +8,38 @@ var wUtils = require('website.utils');
 var _t = core._t;
 
 options.registry.Channel = options.Class.extend({
+    publicChannels: [],
+
+    /**
+     * @override
+     */
+    start() {
+        const prom = this._getPublicChannels().then(channels => {
+            this.publicChannels = channels;
+        });
+        return Promise.all([this._super(...arguments), prom]);
+    },
     /**
      * @override
      */
     cleanForSave: function () {
         this.$target.addClass('d-none');
     },
+    /**
+     * when a new snippet dropped from d&d menu :
+     * if we have already created channels => select the first one
+     * else => modal prompt (create a new channel)
+     *
+     * @override
+     */
+    onBuilt() {
+        if (this.publicChannels.length) {
+            this.$target.attr("data-id", +this.publicChannels[0][0]);
```

Comment:
- No need for the `+`, you are writing a string
- `this.$target[0].dataset.id = ...` is prettier ;)